### ISSUE TITLE
Scipy minimize using analytical gradients

### ISSUE THREAD

**Issue by ProxyCausal** [pymc-labs/pymc-marketing#780]
I'm wondering why in the budget optimizer analytical gradients are not used instead of finite difference since they are faster and more accurate. With large dataset finite difference doesn't seem feasible. It seems gradients for adstock and hill isn't hard to calculate in analytical form.

**Comment by juanitorduz** [comment]
@ProxyCausal would you like to try opening a PR? :)

**Comment by wd60622** [comment]
Should be able to make use of pytensor for the derivatives. pt.grad is the function?

**Comment by wd60622** [comment]
@cetagostini 
Is this something we can use in latest PR?

**Comment by cetagostini** [comment]
Yes, we have a discussion about it on retreat. Let's bring that back, if we have a way to be able to estimate derivatives for all current saturations and adstock, then it would be a nice switch from symbolic to analytical.
